android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -294,3 +295,36 @@ fun Composition.retrieveCompositionSections(): List<Composition.SectionComponent
 fun String.resourceClassType(): Class<out Resource> {
   return Class.forName("org.hl7.fhir.r4.model.$this") as Class<out Resource>
 }
+
+/** Append SyncStrategy tags when it exist. */
+fun Resource.addMandatoryTags(
+  syncStrategy: Map<String, List<String>>,
+) {
+  if (syncStrategy.containsKey(SyncStrategy.CARE_TEAM.value))
+    this.meta.addTag(
+      SyncStrategy.CARE_TEAM.tag.apply {
+        code = syncStrategy.getValue(SyncStrategy.CARE_TEAM.value).first()
+      }
+    )
+
```

Comment:
This code is repeated below. I think we can refactor this to be more readable and avoid the many if statements. You can have the mandatory strategies in a list. Then iterate through the list with only one if statement needed.

Reply:
Sure, make sense